File: frontend/app/types/scorecard.ts
export type SustainabilityDimension =
  | "Economic Performance"
  | "Circular Performance"
  | "Environmental Performance"
  | "Social Performance";

export type QuestionnaireRow = {
  sdg_number: number | null;
  sustainability_dimension: SustainabilityDimension | null;
  sector: string | null;
  score: number | null;

  sdg_description?: string | null;
  sdg_target?: string | null;
  kpi?: string | null;
  question?: string | null;
  score_description?: string | null;
  source?: string | null;
  notes?: string | null;
  status?: string | null;
  comment?: string | null;
};

export type SectorData = Record<string, { rows: QuestionnaireRow[] }>;

File: backend/models/scorecard.py
# backend/models/scorecard.py
from typing import Optional, Dict, List, Literal
from pydantic import BaseModel, Field

class QuestionnaireRow(BaseModel):
    sdg_number: Optional[int] = Field(None, ge=1, le=17)
    sdg_description: Optional[str] = None
    sector: Optional[str] = None
    sdg_target: Optional[str] = None
    sustainability_dimension: Optional[str] = None
    kpi: Optional[str] = None
    question: Optional[str] = None
    score: Optional[int] = Field(None, ge=0, le=5)  # Fixed: was ScoreType
    score_description: Optional[str] = None
    source: Optional[str] = None
    notes: Optional[str] = None
    status: Optional[str] = None
    comment: Optional[str] = None

class SectorRows(BaseModel):
    rows: List[QuestionnaireRow]

class UploadExcelResponse(BaseModel):
    success: bool = True
    data: Dict[str, SectorRows]

class MultiSheetScorecard(BaseModel):
    sheets: Dict[str, SectorRows]  # Each sheet name maps to its respective rows

    def add_sheet(self, sheet_name: str, rows: List[QuestionnaireRow]):
        if sheet_name not in self.sheets:
            self.sheets[sheet_name] = SectorRows(rows=rows)

    def get_sheet(self, sheet_name: str) -> Optional[SectorRows]:
        return self.sheets.get(sheet_name)